### ISSUE TITLE
"Coerce" calls always on option, does it correct behaviour?

### ISSUE THREAD

**Issue by lis355** [yargs/yargs#2158]
Run this simple script WITHOUT any cli arguments:

```javascript
const path = require("path");

const yargs = require("yargs/yargs");

const argv = yargs(process.argv.slice(2))
	.option("someArg", {
		type: "string",
		coerce: path.resolve
	})
	.parse();

```

You will get error message in output:

```
[C:\Program]() Files\nodejs\node.exe [.\main.js]()
Options:
  --help     Show help                                                 [boolean]
  --version  Show version number                                       [boolean]
  --someArg                                                             [string]
node_modules/yargs/build/index.cjs:1
The "path" argument must be of type string. Received undefined
node_modules/yargs/build/index.cjs:1
Process exited with code 1
```

Does it correct? If my someArg is not a demand, why he "gets" default value undefined and then coerce is called? And, of cource, path.resolve drop this error 

I think, it's not correct, to call coerce, if user not give an argument

**Comment by xmedeko** [comment]
Seems like duplicate of #2130

**Comment by lis355** [comment]
@xmedeko Yeah, thanks

**Comment by xmedeko** [comment]
@lis355 If you agree it's a duplicate, then close the issue...

**Comment by xmedeko** [comment]
Should be fixed in 17.4.1.

**Comment by lis355** [comment]
Thanks!

Close because duplicate of https://github.com/yargs/yargs/issues/2130
